app/models/action/edition_diff_factory.rb

```diff
@@ -0,0 +1,45 @@
+class Action::EditionDiffFactory
+  attr_reader :previous_draft_edition, :updated_draft_edition
+
+  def add_previous_draft_edition(previous_draft_edition)
+    @previous_draft_edition = get_edition_attributes(previous_draft_edition)
+  end
+
+  def add_updated_draft_edition(updated_draft_edition)
+    @updated_draft_edition = get_edition_attributes(updated_draft_edition)
+  end
+
+  def create_draft_edition_diff
+    # we don't raise an error if there is no @previous_draft_edition present,
+    # as it's possible for no previous draft edition to exist
+    raise Action::NoUpdatedDraftEditionProvidedError unless @updated_draft_edition.present?
+    create_edition_diff(previous_draft_edition, updated_draft_edition)
+  end
+
+private
+
+  def create_edition_diff(edition_a, edition_b)
+    HashDiff.diff(edition_a, edition_b)
+  end
+
+  def get_edition_attributes(edition)
+    attributes = edition.try(:attributes)
+    return unless attributes
+
+    if edition.links.present?
+      attributes["links"] = []
+      edition.links.each_with_object({}) do |link, hash|
+        hash["link_type"] = link["link_type"]
+        hash["target_content_id"] = link["target_content_id"]
+        attributes["links"] << hash
+      end
+    end
+
+    if edition.change_note.present?
+      attributes["change_note"] = edition.change_note.note
+    end
+    attributes
+  end
+end
+
+class Action::NoUpdatedDraftEditionProvidedError < StandardError; end
```

Comment:
Be good to move this within the class, then it's within autoload scope. Might also be good to have it inherit from RuntimeError to be more semantic (http://blog.honeybadger.io/ruby-exception-vs-standarderror-whats-the-difference/#the-exception-tree)